src/data/lex.rs

```diff
@@ -140,17 +163,31 @@ pub enum ComparisonToken {
     GreaterEqual,
 }
 
-#[derive(Clone, Debug, PartialEq)]
-#[cfg_attr(test, derive(Arbitrary))]
-pub enum Literal {
+#[derive(Clone, Debug)]
+pub enum LiteralToken {
     // literals
-    Int(i64),
-    UnsignedInt(u64),
-    Float(f64),
-    Str(Vec<u8>),
-    Char(u8),
+    Int(RcStr),
+    UnsignedInt(RcStr),
+    Float(RcStr),
+    Str(Vec<RcStr>),
+    Char(RcStr),
 }
 
+impl PartialEq for LiteralToken {
```

Comment:
The options for comparison are:
* Exact pointer location matches
* Source strings match (this implementation)
* The literal is parsed and we see if this is valid
Which is preferred? I'm not entirely sure how it is used.

Reply:
Cool cool